### PR TITLE
Update Scalar DL compatibility test matrix

### PR DESCRIPTION
**Pull request by scalar-boney**
Scalar DL compatibility test matrix updated with the following test results.
|Scalar DL Ledger Version   |Scalar DL Auditor Version  |Scalar DL Java Client SDK Version  |
|---------------------------|---------------------------|-----------------------------------|
|3.4.1  |3.4.1   |3.0.5, 3.1.5, 3.2.5, 3.3.4, 3.4.1  |
|3.3.4  |3.3.4   |3.0.5, 3.1.5, 3.2.5, 3.3.4, 3.4.1   |
|3.2.5  |3.2.5   | 3.0.5, 3.1.5, 3.2.5, 3.3.4, 3.4.1  |
|3.1.5  |3.1.5   |3.0.5, 3.1.5, 3.2.5, 3.3.4, 3.4.1   |
|3.0.5  |3.0.5   |3.0.5, 3.1.5, 3.2.5, 3.3.4, 3.4.1   |

Current Scalar DL Java Client SDK tool applicable only for Scalar DL version 3.0 or later.

Scalar DL Ledger and Auditor 3.0.5 were tested with Scalar DL schema loader 3.1.4 because Scalar DL schema loader 3.0.0 does not support the latest schema loader options.

### CODE REVIEW

**Comment by feeblefakie** [comment]
@scalar-boney Can you check with the current master? The master is going to be 3.5, and I would like to know if the master is backward compatible before release.

**Comment by scalar-boney** [comment]
> @scalar-boney Can you check with the current master? The master is going to be 3.5, and I would like to know if the master is backward compatible before release.

Scalar DL compatibility test matrix updated with the following test results.
|Scalar DL Ledger Version   |Scalar DL Auditor Version  |Scalar DL Java Client SDK Version  |
|---------------------------|---------------------------|-----------------------------------|
|fix-function-extraction (Scalar branch)  |fix-function-extraction  (Scalar branch) |fix-function-extraction  (Scalar branch)  |
|master  (Scalar commit id  d53cabe)|master  (Scalar commit id d53cabe)   |3.0.5, 3.1.5, 3.2.5, 3.3.4, 3.4.1   |

Scalar DL schema loader 3.4.1 used.

File: docs/compatibility.md [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-|2.0.8         |2.0.9 - 2.1.0                |
-|2.0.7         |2.0.0 - 2.0.8                |
-|2.0.6         |2.0.0 - 2.0.8                |
-|2.0.5         |2.0.0 - 2.0.8                |
-|2.0.4         |2.0.0 - 2.0.8                |
-|2.0.3         |2.0.0 - 2.0.8                |
-|2.0.2         |2.0.0 - 2.0.8                |
-|2.0.1         |2.0.0 - 2.0.8                |
-|2.0.0         |2.0.0 - 2.0.8                |
+|Scalar DL Ledger Version   |Scalar DL Auditor Version  |Scalar DL Java Client SDK Version  |
```
Comment (feeblefakie) [inline docs/compatibility.md:5]:
```suggestion
|Scalar DL Ledger/Auditor version   |Scalar DL Java Client SDK version  |
```

File: docs/compatibility.md [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
+|:--------------------------|:--------------------------|:----------------------------------|
+|3.5  |3.5   |3.0 - 3.5   |
+|3.4  |3.4   |3.0 - 3.4   |
+|3.3  |3.3   |3.0 - 3.4   |
+|3.2  |3.2   |3.0 - 3.4   |
+|3.1  |3.1   |3.0 - 3.4   |
+|3.0  |3.0   |3.0 - 3.4   |
+|2.2  |Not available   |Not tested  |
```
Comment (feeblefakie) [inline docs/compatibility.md:13]:
```suggestion
|2.2 (Auditor is not available) |Not tested  |
```

**Review by feeblefakie — COMMENTED** [review]
Thanks!
Left some suggestions on the format.

File: docs/compatibility.md (hunk 2 above)
Comment (scalar-boney) [reply docs/compatibility.md]: Matrix updated with the new format.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Comment by feeblefakie** [comment]
Please don't merge it yet.

**Pull request merged**